### PR TITLE
Add ProcessorCount Event to RuntimeEventSource

### LINKED ISSUE #56143: [Feature Request]: Diagnostics for processor count
Original combined issue [here](https://github.com/dotnet/runtime/issues/56072)

The goal is to help customers better understand how their applications are running by adding events and counters that can give more insights and will help them debug potential performance issues. It'll also help us debug customer applications and write tools that would be able to glean insights from traces.

What was the value of Environment.ProcessorCount should be emitted in an event. I'm not sure which event is most appropriate for this (if we have an existing one) but we need a way to get the runtime settings.

### PR DESCRIPTION
**Pull request by mikelle-rogers**
fixes #56143

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs [hunk 1]
```diff
@@ -66,6 +68,12 @@ internal void LogAppContextSwitch(string switchName, int value)
             base.WriteEvent((int)EventId.AppContextSwitch, switchName, value);
         }
 
+        [Event((int)EventId.ProcessorCount, Level = EventLevel.Informational, Keywords = Keywords.NumProcesses)]
+        internal void ProcessorCount(int processorCount)
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs:72]: What calls this when?

Comment (noahfalk) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs:72]: I assume the PR is incomplete. I'd recommend we call it inside OnEventCommand() when the EventSource is being enabled, similar to the LogSwitchValues() down on line 115.

**Review by davmason — APPROVED** [review]
LGTM

**Review by josalem — APPROVED** [review]
Could we add a test for this event?

e.g., `Assert.Equal(Environment.ProcessorCount, countFromEvent);`

**Review by davmason — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by tommcdon** [comment]
Adding @brianrob  in case TraceEvent changes are needed.

**Comment by brianrob** [comment]
@tommcdon, thanks for including me here.  This doesn't require a change to TraceEvent because we don't have a parser for this provider - it is emitted via the standard `EventSource` mechanism.

This does bring up a question for me.  Historically, if I understand correctly, this source was used exclusively for event counters.  I see that now the source also logs app context switches, and now the processor count.  How should I think about the split between this provider and the Microsoft-Windows-DotNetRuntime provider?

**Pull request merged**